internal/applicationsnapshot/report_test.go

```diff
@@ -151,7 +151,62 @@ policy:
 	assert.YAMLEq(t, expected, string(reportYaml))
 	assert.False(t, report.Success)
 }
+func Test_ReportMarkdownSummary(t *testing.T) {
+	cases := []struct {
+		name       string
+		expected   string
+		snapshot   string
+		components []Component
+	}{
+		{
+			name: "One Warning and One Success",
+			expected: `| Field     | Value |Status|
+|-----------|-------|-------|
+| Time | 1970-01-01 00:00:00 |  |
+| Name |   |  |
+| Successes | 1 | :x: |
+| Failures | 0 | :white_check_mark: |
+| Warnings | 1 | :x: |
+| Result | true | :white_check_mark: |` + "\n",
+			components: []Component{
+				{
+					Successes: []evaluator.Result{
+						{Message: "Success1"},
+					},
+					Warnings: []evaluator.Result{
+						{Message: "Warning1"},
+					},
+					Success: true,
+				},
+			},
+		},
+	}
+	for _, c := range cases {
+		t.Run(c.name, func(t *testing.T) {
+			fs := afero.NewMemMapFs()
+			defaultWriter, err := fs.Create("default")
+			assert.NoError(t, err)
+
+			ctx := context.Background()
+			report, err := NewReport(c.snapshot, c.components, createTestPolicy(t, ctx), nil, nil)
```

Comment:
Could invoke `generateMarkdownSummary` directly here and use github.com/gkampitakis/go-snaps, like [here](https://github.com/enterprise-contract/ec-cli/blob/5526af19d1e6eabd3d24c3ecf864c8ed6646ff0a/internal/signature/certificate_test.go#L55)